kiosk/src/Models/Kiosk.ts

```diff
@@ -52,11 +50,13 @@ export class Kiosk {
         this.addNewGamesToList();
     }
 
-    saveNewGame(gameId: string): GameData | undefined {
+    async saveNewGame(gameId: string): Promise<GameData | undefined> {
         const allAddedGames = this.getAllAddedGames();
         if (!allAddedGames[gameId]) {
-            this.numAddedGames += 1;
-            const newGame = new GameData(gameId, 'Kiosk Game', this.addedGameDescription, "None");
+            const gameName = await getGameDetail(gameId, "name");
+            const gameDescription = await getGameDetail(gameId, "description");
```

Comment:
you don't need to make the same request twice. can't you do it once and get both the name and description in one go?